pghoard/pgutil.py

```diff
@@ -92,3 +93,14 @@ def parse_connection_string_libpq(connection_string):
                 value, connection_string = rem, ""
         fields[key] = value
     return fields
+
+
+def check_if_pg_connection_is_alive(db_conn) -> bool:
+    if db_conn.closed:
+        return False
+
+    status = db_conn.get_transaction_status()
+    if status not in [TRANSACTION_STATUS_ACTIVE, TRANSACTION_STATUS_IDLE]:
+        return False
```

Comment:
Shouldn't we include TRANSACTION_STATUS_INTRANS as well ?
Not sure it matters here as we shouldn't be in a transaction but still :-)